---
title: "BIRC6 RNAseq"
output:
  html_document:
    toc: true
    toc_float: true
    theme: spacelab
toc_float: true
editor_options: 
  chunk_output_type: console
---
    
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE)
library(readr); 
library(taigr)
library(stringr)
library(plyr); 
library(tibble); 
library(magrittr); 
library(tidyr); 
library(dplyr); 
#library(dependr)
library(ggplot2)
library(ggrepel)
library(edgeR)
library(limma)
library(knitr)
library(DT)
library(cdsr)
library(GSEABase)
library(RColorBrewer)
library(pheatmap)
library(cowplot)

data_dir <- 'BIRC6_RNAseq_analysis/data/'

q_thresh <- 0.1
results_dir <- './BIRC6_RNAseq_analysis/results'

# full_gsc <- read_rds('~/CPDS/data/MSigDB/gsc_data.rds')
full_gsc <- read_rds(download.raw.from.taiga(data.name='msigdb-gene-set-collections-8453', data.version=2))
```

```{r, include = FALSE}
sample_info <- read.csv(file.path(data_dir, 'BIRC6_RNAseq_sample_list.csv'), stringsAsFactors = F, check.names = F) %>% 
  dplyr::rename(sample = `sample name`,
                cell_line = `cell line`,
                dep = `dep/indep`) %>% 
  mutate(perturbation = make.names(perturbation))

counts_mat <- read_tsv(file.path(data_dir, 'RNAseq_20190104.rsem_genes_expected_count.txt.gz')) %>%
  as.data.frame() %>% 
  dplyr::select(-`transcript_id(s)`) %>% 
    column_to_rownames(var = 'gene_id') 

tpm_mat <- read_tsv(file.path(data_dir, 'RNAseq_20190104.rsem_genes_tpm.txt.gz')) %>%
  as.data.frame() %>% 
  dplyr::select(-`transcript_id(s)`) %>% 
    column_to_rownames(var = 'gene_id') 
  
```


```{r}
remap_genes <- FALSE
gene_info_file <- paste0(results_dir, '/gene_mapping_info.rds')
if (!remap_genes & file.exists(gene_info_file)) {
  gene_df <- read_rds(gene_info_file)
} else {
  library('biomaRt')
  mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
  genes <- rownames(counts_mat)
  gene_ids <- str_match(genes, '(.+)\\.[0-9]+$')[,2]
  G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id_version", "ensembl_gene_id", "hgnc_symbol"), values=gene_ids,mart= mart)
  
  gene_df <- G_list[match(gene_ids, G_list$ensembl_gene_id),]
  gene_df %<>% mutate(gene_id = genes)
  write_rds(gene_df, gene_info_file)
}
```


```{r}
counts_mat_lane_col <- laply(sample_info$sample, function(cur_samp) {
  cur_set <- grepl(cur_samp, colnames(counts_mat))
  rowSums(counts_mat[, cur_set])
}) %>% set_rownames(sample_info$sample) %>% t()

tpm_lane_col <- laply(sample_info$sample, function(cur_samp) {
  cur_set <- grepl(cur_samp, colnames(tpm_mat))
  rowMedians(as.matrix(tpm_mat[, cur_set]))
}) %>% set_rownames(sample_info$sample) %>% set_colnames(rownames(tpm_mat)) %>% t()

dge <- DGEList(counts_mat_lane_col)

#make sure the cond_table and counts_mat are properly aligned
stopifnot(all.equal(sample_info$sample, colnames(dge))) 

dge$genes <- gene_df

```


```{r}
min_cpm <- 1
min_group_size <- ncol(counts_mat_lane_col)/14

lib_sizes <- colSums(counts_mat_lane_col)
paste0('Max fold diff lib size = ', 10 ^ diff(range(log10(lib_sizes))))

sample_info %<>% mutate(library_size = lib_sizes)
ggplot(sample_info, aes(x = cell_line,y=library_size)) + 
  ggbeeswarm::geom_beeswarm(aes(color = cell_line), size = 3, alpha = 0.75) +
  theme(axis.text.x = element_text(angle = 75, hjust = 1))

CPM <- cpm(dge)
keep <- rowSums(CPM > min_cpm) >= min_group_size
paste0(nrow(dge), ' total transcripts. Keeping ', sum(keep), ' transcripts after filtering')
dge <- dge[keep,]
dge <- edgeR::calcNormFactors(dge, method = 'TMM') #scale normalization using TMM
logCPM <- cpm(dge, log=TRUE)

```

# QC checks

## Heirarchical clustering

Heirarchical clustering based on expression profile Spearman correlations. This shows that the data are overall high quality (no outliers), with good grouping of samples within-cell lines. The biggest difference is clearly between colon and ovarian cell lines.

```{r, fig.height=8, fig.width=8}
# sampleDists <- as.matrix(dist(t(logCPM)))
sampleDists <- as.matrix(cor(logCPM, method = 'spearman'))
vector <- sample_info$target_id
mycols <- brewer.pal(8, "Dark2")[1:length(unique(vector))]
pheatmap(sampleDists)
```

## Multi-dimensional scaling

```{r, fig.height=7, fig.width=6}
plotMDS(logCPM)
```

## PCA 

```{r, fig.height=7, fig.width=6}
pc <- prcomp(t(logCPM))
plot(pc$x[, 1:2], pch = 19, cex=1)
text(pc$x[, 1:2], labels = colnames(logCPM), pos = 3)
```

## lCPM distributions

Comparing distribution of log-counts per million per sample looks reasonable

```{r, fig.height=7, fig.width=6}
library(RColorBrewer)
nsamples <- ncol(dge)
col <- brewer.pal(nsamples, "Paired")

plot(density(logCPM[,1]), col=col[1], lwd=2, ylim=c(0,0.21), las=2,
 main="", xlab="")
title(main="B. Filtered data", xlab="Log-cpm")
abline(v=0, lty=3)
for (i in 2:nsamples){
 den <- density(logCPM[,i])
 lines(den$x, den$y, col=col[i], lwd=2)
}
```

## Replicate correlations

```{r}
sample_info %<>% mutate(samp_pert = paste0(cell_line, '.', perturbation))
rep_cors <- ldply(unique(sample_info$samp_pert), function(cur_pert) {
  cur_set <- sample_info %>% filter(samp_pert == cur_pert) %>% .[['sample']]
  stopifnot(length(cur_set) == 2)
  cc <- cor(logCPM[,cur_set[1]], logCPM[, cur_set[2]])
  data.frame(cor = cc,
             samp_pert = cur_pert)
}) %>% mutate(cl = str_match(samp_pert, '(.+)\\.sg')[,2])

ggplot(rep_cors, aes(x = cl,y=cor)) + 
  ggbeeswarm::geom_beeswarm(aes(color = cl), size = 3, alpha = 0.75) +
  theme(axis.text.x = element_text(angle = 75, hjust = 1)) +
  ylab('Replicate correlation')

```


## Comparing BIRC6 expression levels

```{r}
targ_gene <- which(dge$genes$hgnc_symbol == 'BIRC6')
df <- data.frame(BIRC6_GE = logCPM[targ_gene,], sample_name = colnames(logCPM)) %>% 
  cbind(sample_info) %>% 
  dplyr::group_by(samp_pert, cell_line, perturbation) %>% 
  dplyr::summarise(BIRC6_GE = mean(BIRC6_GE))

head(df)

ggplot(df, aes(cell_line, BIRC6_GE, fill = perturbation)) + 
  geom_bar(stat = 'identity', position = 'dodge') +
  ylab('BIRC6 expression (avg log CPM)') + 
  xlab('') +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

Collapse replicates

```{r}
unique_samps <- as.character(unique(sample_info$samp_pert))
counts_rep_col <- laply(unique_samps, function(cur_samp) {
  cur_set <- which(as.character(sample_info$samp_pert) == cur_samp)
  rowSums(counts_mat_lane_col[keep, cur_set])
}) %>% set_rownames(unique_samps) %>% t()

tpm_rep_col <- laply(unique_samps, function(cur_samp) {
  cur_set <- which(as.character(sample_info$samp_pert) == cur_samp)
  rowMedians(tpm_lane_col[keep, cur_set])
}) %>% set_rownames(unique_samps) %>% set_colnames(rownames(tpm_lane_col[keep,])) %>% t()

sample_info_rep_col <- sample_info %>% 
  mutate(samp_pert = as.character(samp_pert)) %>% 
  distinct(samp_pert, .keep_all=T)

dge_rep_col <- DGEList(counts_rep_col)

#make sure the cond_table and counts_mat are properly aligned
stopifnot(all.equal(sample_info_rep_col$samp_pert, colnames(dge_rep_col))) 

dge_rep_col$genes <- gene_df[match(rownames(dge_rep_col), gene_df$gene_id),]

```

Save LCPM data (rep collapsed)
```{r}
logCPM <- cpm(dge_rep_col, log=TRUE)

cur_gene_df <- gene_df[match(rownames(logCPM), gene_df$gene_id),] %>% 
  mutate(gene_str = paste0(hgnc_symbol, ' (', gene_id, ')'))

write.csv(logCPM %>% set_rownames(cur_gene_df$gene_str), './BIRC6_RNAseq_analysis/results/BIRC6_rep_collapsed_logCPM.csv')
```

<!-- Compute ssGSEA -->

<!-- ```{r} -->
<!-- write.csv(tpm_rep_col %>% set_rownames(cur_gene_df$gene_str), './BIRC6_RNAseq_analysis/results/BIRC6_rep_collapsed_TPM.csv') -->

<!-- cur_hgnc <- gene_df[match(rownames(tpm_rep_col), gene_df$gene_id),'hgnc_symbol'] -->
<!-- tpm_mat_hgnc <- tpm_rep_col %>% set_rownames(cur_hgnc) -->
<!-- tpm_mat_hgnc <- tpm_mat_hgnc[!is.na(cur_hgnc),] -->

<!-- hallmark_ssGSEA = GSVA::gsva((tpm_mat_hgnc), full_gsc$hallmark, method = 'ssgsea') -->
<!-- canonical_ssGSEA = GSVA::gsva((tpm_mat_hgnc), full_gsc$canonical, method = 'ssgsea') -->
<!-- GO_BP_ssGSEA = GSVA::gsva((tpm_mat_hgnc), full_gsc$GO_biological_process, method = 'ssgsea') -->

<!-- write.csv(hallmark_ssGSEA, './BIRC6_RNAseq_analysis/results/BIRC6_ssGSEA_hallmark.csv') -->
<!-- write.csv(canonical_ssGSEA, './BIRC6_RNAseq_analysis/results/BIRC6_ssGSEA_canonical.csv') -->
<!-- write.csv(GO_BP_ssGSEA, './BIRC6_RNAseq_analysis/results/BIRC6_ssGSEA_GO_BP.csv') -->
<!-- ``` -->


```{r}
#helper function for gsea
run_gsea_analysis <- function(limma_fit, cur_contr, cur_gsc, nperm = 1e6) {
  gsea_save_dir <- './BIRC6_RNAseq_analysis/results/GSEA_stats'
  
  cur_res <- topTable(limma_fit, number = Inf, coef = cur_contr) %>% 
   filter(hgnc_symbol != "BIRC6", hgnc_symbol != '', !is.na(hgnc_symbol)) %>% 
    distinct(hgnc_symbol, .keep_all=TRUE)
  
  LFC_vals <- cur_res$logFC %>% set_names(cur_res$hgnc_symbol)
  
  fgsea_res <- cdsr::run_fGSEA(full_gsc[[cur_gsc]], gene_stat = LFC_vals, perm_type = 'gene', nperm = nperm, nproc = 6) %>% 
    arrange(pval)
  
  g <- ggplot(fgsea_res, aes(NES, -log10(pval), color = padj < 0.1)) + 
    # geom_jitter(alpha = 0.75, width = 0.05, height = 0.05, size = 2) + 
    geom_point(alpha = 0.6, size = 1.5) + 
    geom_text_repel(data = fgsea_res %>% arrange(dplyr::desc(abs(NES))) %>% head(10), aes(label = pathway), size = 1.5) +
    ggtitle(cur_contr) +
    # guides(color = guide_legend(title = element_blank(), nrow = 2)) +
    theme_Publication()
  
  fgsea_res <- fgsea_res %>% 
    mutate(LE_n = lengths(leadingEdge)) %>% 
    mutate(leadingEdge = ifelse(LE_n > 1, str_sub(paste0(leadingEdge), 3, -2), leadingEdge)) %>% 
    mutate(leadingEdge = str_replace_all(leadingEdge, '"', ''))
  
  # write.csv(fgsea_res %>% dplyr::select(-leadingEdge), file.path(gsea_save_dir, paste0(cur_contr, '_', cur_gsc, '.csv')), row.names = FALSE)
  write_csv(fgsea_res %>% dplyr::select(-LE_n), file.path(gsea_save_dir, paste0(cur_contr, '_', cur_gsc, '.csv')))
  return(g)
}
```


# Diff expression analysis

Collapse across replicates by summing read counts
Model cell line as random effect

```{r}
sample_info_rep_col %<>% mutate(perturbation = factor(perturbation),
                                pert_type = paste0(perturbation, '.', dep),
                                pert_type = factor(pert_type))
design <- model.matrix(~0 + pert_type, data = sample_info_rep_col)

sample_info_rep_col$pert_type %<>% as.factor()

colnames(design) <- levels(sample_info_rep_col$pert_type)
```

```{r}
cm <- makeContrasts(
        dep_avg = 0.5*(sgBIRC6.4.dependent + sgBIRC6.1.dependent) - sgCh2.2.dependent,
        ind_avg = 0.5*(sgBIRC6.4.independent + sgBIRC6.1.independent) - sgCh2.2.independent,
        dep_vs_ind = (0.5*(sgBIRC6.4.dependent + sgBIRC6.1.dependent)- sgCh2.2.dependent) - (0.5*(sgBIRC6.4.independent + sgBIRC6.1.independent)- sgCh2.2.independent),
        ov_avg = 0.5*(0.5*(sgBIRC6.4.dependent + sgBIRC6.1.dependent)- sgCh2.2.dependent) + 0.5*(0.5*(sgBIRC6.4.independent + sgBIRC6.1.independent)- sgCh2.2.independent),
        levels = design
    )
v <- voom(dge_rep_col, design, plot=TRUE)
corfit <- duplicateCorrelation(v, design, block=sample_info_rep_col$cell_line)

fit <- lmFit(v, design, block = sample_info_rep_col$cell_line, correlation = corfit$consensus.correlation)
# fit <- lmFit(v, design)
fit2 <- contrasts.fit(fit, cm)
fit2 <- eBayes(fit2)
```

## DE in BIRC6 Dependent lines

```{r}
res_dep <- topTable(fit2, number = Inf, coef = 'dep_avg')

ggplot(res_dep, aes(logFC, -log10(P.Value))) + 
  geom_point(aes(color = adj.P.Val < 0.1)) +
  geom_point(data = res_dep %>% filter(hgnc_symbol == 'BIRC6'),
             color = 'blue', size = 3) +
  geom_text_repel(data = res_dep %>% 
                    filter(adj.P.Val < 0.1) %>% 
                    arrange(desc(abs(logFC))) %>% 
                    head(50),
                  aes(label = hgnc_symbol),
                  size = 2) +
  cdsr::scale_color_Publication() +
  cdsr::theme_Publication()

res_dep %>% 
  dplyr::select(-ensembl_gene_id_version, -ensembl_gene_id) %>% 
  write.csv('./BIRC6_RNAseq_analysis/results/DE_res_dep.csv', row.names = F)
```

### GSEA
```{r}
run_gsea_analysis(fit2, 'dep_avg', 'hallmark')
run_gsea_analysis(fit2, 'dep_avg', 'canonical')
run_gsea_analysis(fit2, 'dep_avg', 'GO_biological_process')
```


## DE in BIRC6 Independent lines

```{r}
res_ind <- topTable(fit2, number = Inf, coef = 'ind_avg')

ggplot(res_ind, aes(logFC, -log10(P.Value))) + 
  geom_point(aes(color = adj.P.Val < 0.1)) +
  geom_point(data = res_ind %>% filter(hgnc_symbol == 'BIRC6'),
             color = 'blue', size = 3) +
  geom_text_repel(data = res_ind %>% 
                    filter(adj.P.Val < 0.1) %>% 
                    arrange(dplyr::desc(abs(logFC))) %>% 
                    head(50),
                  aes(label = hgnc_symbol),
                  size = 2) +
  cdsr::scale_color_Publication() +
  cdsr::theme_Publication()

res_ind %>% 
  dplyr::select(-ensembl_gene_id_version, -ensembl_gene_id) %>% 
  write.csv('./BIRC6_RNAseq_analysis/results/DE_res_ind.csv', row.names = F)

```

### GSEA
```{r}
run_gsea_analysis(fit2, 'ind_avg', 'hallmark')
run_gsea_analysis(fit2, 'ind_avg', 'canonical')
run_gsea_analysis(fit2, 'ind_avg', 'GO_biological_process')
```

## Comparison of DE in Dependent vs Independent lines

Estimated KO effect in Dependent - Independent lines 

```{r}
res_dep_vs_ind <- topTable(fit2, number = Inf, coef = 'dep_vs_ind')

ggplot(res_dep_vs_ind, aes(logFC, -log10(P.Value))) + 
  geom_point(aes(color = adj.P.Val < 0.1)) +
  geom_point(data = res_dep_vs_ind %>% filter(hgnc_symbol == 'BIRC6'),
             color = 'blue', size = 3) +
  geom_text_repel(data = res_dep_vs_ind %>% 
                    filter(adj.P.Val < 0.1) %>% 
                    arrange(dplyr::desc(abs(logFC))) %>% 
                    head(50),
                  aes(label = hgnc_symbol),
                  size = 2) +
  cdsr::scale_color_Publication() +
  cdsr::theme_Publication()

res_dep_vs_ind %>% 
  dplyr::select(-ensembl_gene_id_version, -ensembl_gene_id) %>% 
  write.csv('./BIRC6_RNAseq_analysis/results/DE_res_dep_vs_ind.csv', row.names = F)

```

### GSEA
```{r}
run_gsea_analysis(fit2, 'dep_vs_ind', 'hallmark')
run_gsea_analysis(fit2, 'dep_vs_ind', 'canonical')
run_gsea_analysis(fit2, 'dep_vs_ind', 'GO_biological_process')
```

## Average DE across all cell lines

```{r}
res_ov <- topTable(fit2, number = Inf, coef = 'ov_avg')

ggplot(res_ov, aes(logFC, -log10(P.Value))) + 
  geom_point(aes(color = adj.P.Val < 0.1)) +
  geom_point(data = res_ov %>% filter(hgnc_symbol == 'BIRC6'),
             color = 'blue', size = 3) +
  geom_text_repel(data = res_ov %>% 
                    filter(adj.P.Val < 0.1) %>% 
                    arrange(dplyr::desc(abs(logFC))) %>% 
                    head(50),
                  aes(label = hgnc_symbol),
                  size = 2) +
  cdsr::scale_color_Publication() +
  cdsr::theme_Publication()

res_ov %>% 
  dplyr::select(-ensembl_gene_id_version, -ensembl_gene_id) %>% 
  write.csv('./BIRC6_RNAseq_analysis/results/DE_res_ov.csv', row.names = F)

```


### GSEA
```{r}
run_gsea_analysis(fit2, 'ov_avg', 'hallmark')
run_gsea_analysis(fit2, 'ov_avg', 'canonical')
run_gsea_analysis(fit2, 'ov_avg', 'GO_biological_process')
```


#model individual cell line effects

```{r}
sample_info %<>% mutate(samp_pert = factor(samp_pert))
design <- model.matrix(~0 + samp_pert, data = sample_info)
colnames(design) <- levels(sample_info$samp_pert)
```

```{r}
cm <- makeContrasts(
        SW837 = 0.5*(SW837.sgBIRC6.1 + SW837.sgBIRC6.4) - SW837.sgCh2.2,
        HCC15 = 0.5*(HCC15.sgBIRC6.1 + HCC15.sgBIRC6.4) - HCC15.sgCh2.2,
        HCC202 = 0.5*(HCC202.sgBIRC6.1 + HCC202.sgBIRC6.4) - HCC202.sgCh2.2,
        HCC95 = 0.5*(HCC95.sgBIRC6.1 + HCC95.sgBIRC6.4) - HCC95.sgCh2.2,
        JIMT1 = 0.5*(JIMT1.sgBIRC6.1 + JIMT1.sgBIRC6.4) - JIMT1.sgCh2.2,
        SNU503 = 0.5*(SNU503.sgBIRC6.1 + SNU503.sgBIRC6.4) - SNU503.sgCh2.2,
        levels = design
    )
v <- voom(dge, design, plot=TRUE)
corfit <- duplicateCorrelation(v, design, block=sample_info$`replicate #`)

fit <- lmFit(v, design, block = sample_info$`replicate #`, correlation = corfit$consensus.correlation)
# fit <- lmFit(v, design)
fit2 <- contrasts.fit(fit, cm)
fit2 <- eBayes(fit2)
```

```{r}
all_res <- ldply(colnames(cm), function(cur_cont) {
    topTable(fit2, coef = cur_cont, number = Inf) %>% 
        rownames_to_column(var = 'Gene') %>% 
        mutate(contrast = cur_cont)
})

all_res %>% 
  dplyr::select(cell_line = contrast, Gene, hgnc_symbol, logFC, AveExpr, P.Value, adj.P.Val) %>% 
  write.csv('./BIRC6_RNAseq_analysis/results/all_DE_results.csv', row.names=F)
```

```{r}
LFC_mats <- all_res %>% 
    dplyr::select(hgnc_symbol, logFC, contrast) %>% 
  distinct(hgnc_symbol, contrast, .keep_all=T) %>% 
    reshape2::acast(hgnc_symbol ~ contrast, value.var = 'logFC')

# C <- cor(LFC_mats, method = 'pearson')
# diag(C) <- NA
# pheatmap(C)

top_hits <- res_ov %>% 
  filter(hgnc_symbol %in% rownames(LFC_mats), hgnc_symbol != '') %>% 
  filter(adj.P.Val < 0.05) %>% 
  # arrange(desc(abs(logFC))) %>% 
  # head(1000) %>% 
  .[['hgnc_symbol']]
C <- cor(LFC_mats[top_hits,], method = 'pearson')
diag(C) <- NA
pheatmap(C)

```

```{r, fig.width=8, fig.height=6}
make_volcano <- function(cur_contr) {
to_plot <- filter(all_res, contrast == cur_contr)
ggplot(to_plot,
       aes(logFC, -log10(P.Value), color = adj.P.Val < q_thresh)) + 
  geom_point(alpha = 0.5) + 
  geom_point(data = filter(to_plot, Gene == 'WRN'), color = 'darkgreen', alpha = 1, size = 3) + 
  geom_text_repel(data = to_plot %>% arrange(desc(abs(logFC))) %>% head(30), aes(label = hgnc_symbol)) +
  geom_vline(xintercept = 0, linetype = 'dashed') + 
  ggtitle(cur_contr)
}

for (cur_CL in colnames(cm)) {
  make_volcano(cur_CL)
ggsave(file.path('./BIRC6_RNAseq_analysis/results/', paste0(cur_CL, '.png')), width = 8, height = 5)
}
```

### GSEA
```{r}
for (cur_CL in colnames(cm)) {
  run_gsea_analysis(fit2, cur_CL, 'hallmark')
  run_gsea_analysis(fit2, cur_CL, 'canonical')
  run_gsea_analysis(fit2, cur_CL, 'GO_biological_process')
}
```

```{r}
all_gsea <- ldply(colnames(cm), function(cur_CL) {
  read_csv(file.path('./BIRC6_RNAseq_analysis/results/GSEA_stats', paste0(cur_CL, '_hallmark.csv'))) %>% 
    mutate(CCLE_ID = cur_CL)
})
GSEA_NES <- reshape2::acast(all_gsea, pathway ~ CCLE_ID, value.var = 'NES')
pheatmap::pheatmap(GSEA_NES, fontsize_row = 6)
```



```{r}
# HCC <- read_csv('~/CPDS/jmm-scratch/BIRC6/results/GSEA_stats/HCC202_GO_biological_process.csv') %>% dplyr::select(-leadingEdge)
# SNU <- read_csv('~/CPDS/jmm-scratch/BIRC6/results/GSEA_stats/SNU503_GO_biological_process.csv') %>% dplyr::select(-leadingEdge)
HCC <- read_csv('./BIRC6_RNAseq_analysis/results/GSEA_stats/HCC202_hallmark.csv') %>% dplyr::select(-leadingEdge)
SNU <- read_csv('./BIRC6_RNAseq_analysis/results/GSEA_stats/SNU503_hallmark.csv') %>% dplyr::select(-leadingEdge)
# HCC <- read_csv('~/CPDS/jmm-scratch/BIRC6/results/GSEA_stats/HCC202_canonical.csv') %>% dplyr::select(-leadingEdge)
# SNU <- read_csv('~/CPDS/jmm-scratch/BIRC6/results/GSEA_stats/SNU503_canonical.csv') %>% dplyr::select(-leadingEdge)


comb <- full_join(HCC, SNU, by = 'pathway', suffix = c('_HCC202', '_SNU503')) 
comb %<>% mutate(pathway = str_replace(pathway, 'HALLMARK_', ''))

comb %<>% mutate(avg_p = 0.5*(-log10(pval_HCC202) - log10(pval_SNU503)))
ggplot(comb, aes(NES_SNU503, NES_HCC202)) + 
  geom_point(aes(size = avg_p), alpha = 0.5) +
  geom_abline(linetype = 'dashed') + 
  geom_hline(yintercept = 0, linetype = 'dashed') + 
  geom_vline(xintercept = 0, linetype = 'dashed') +
  scale_size_continuous(range = c(1, 5)) +
  guides(size = guide_legend(title = 'Avg -log10(pval)')) +
  geom_text_repel(data = comb %>% 
                    mutate(rank = pmax(abs(NES_SNU503), abs(NES_HCC202))) %>% 
                    arrange(dplyr::desc(rank)) %>% 
                    head(10),
                  aes(label = pathway),
                  size = 2.5)
ggsave('./BIRC6_RNAseq_analysis/results/HCC_SNU_GSEA_scatter_hallmark.png', width = 6, height = 4)
ggsave('./BIRC6_RNAseq_analysis/results/HCC_SNU_GSEA_scatter_hallmark.pdf', width = 6, height = 4)
```

